refarch-apigateway/apigateway/src/main/java/refarch/configuration/WebSessionConfiguration.java

```diff
@@ -0,0 +1,117 @@
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.oauth2.client.web.server.ServerOAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.server.WebSessionServerOAuth2AuthorizedClientRepository;
+import org.springframework.session.MapSession;
+import org.springframework.session.ReactiveMapSessionRepository;
+import org.springframework.session.ReactiveSessionRepository;
+import org.springframework.session.Session;
+import org.springframework.session.config.annotation.web.server.EnableSpringWebSession;
+import org.springframework.session.hazelcast.HazelcastIndexedSessionRepository;
+
+/**
+ * This class configures Hazelcast as the ReactiveSessionRepository.
+ *
+ */
+@Configuration
+@EnableSpringWebSession
+public class WebSessionConfiguration {
+
+    @Value("${hazelcast.instance:hazl_instance}")
+    public String hazelcastInstanceName;
+
+    @Value("${hazelcast.group-name:session_replication_group}")
+    public String groupConfigName;
+
+    @Value("${app.spring-session-hazelcast.namespace:my_namespace}")
+    public String openshiftNamespace;
+
+    @Value("${hazelcast.openshift-service-name:apigateway}")
+    public String openshiftServiceName;
+
+    @Bean
+    public ServerOAuth2AuthorizedClientRepository authorizedClientRepository() {
+        return new WebSessionServerOAuth2AuthorizedClientRepository();
+    }
+
+    @Bean
+    public ReactiveSessionRepository<MapSession> reactiveSessionRepository(@Qualifier("hazelcastInstance") @Autowired HazelcastInstance hazelcastInstance) {
+        final IMap<String, Session> map = hazelcastInstance.getMap(HazelcastIndexedSessionRepository.DEFAULT_SESSION_MAP_NAME);
+        return new ReactiveMapSessionRepository(map);
+    }
+
+    @Bean
+    @Profile({ "local", "test" })
+    public Config localConfig(@Value(
+        "${spring.session.timeout}"
+    ) int timeout) {
+        final var hazelcastConfig = new Config();
+        hazelcastConfig.setInstanceName(hazelcastInstanceName);
+        hazelcastConfig.setClusterName(groupConfigName);
+
+        addSessionTimeoutToHazelcastConfig(hazelcastConfig, timeout);
+
+        final var networkConfig = hazelcastConfig.getNetworkConfig();
+
+        final var joinConfig = networkConfig.getJoin();
+        joinConfig.getMulticastConfig().setEnabled(false);
+        joinConfig.getTcpIpConfig()
+                .setEnabled(true)
+                .addMember("127.0.0.1");
+
+        return hazelcastConfig;
+    }
+
+    @Bean
+    @Profile({ "dev", "kon", "prod" })
```

Comment:
References to those profiles should be removed. Maybe we can set this as the default? (And thus for not running locally)?

Reply:
crated issue #38